### DISCUSSION TITLE
feat(manager/asdf): Add even more ASDF plugins (yq/jq/gh/sbt/...)

### DISCUSSION THREAD

**Discussion by MPV** [renovatebot/renovate#24582] [Suggest an Idea]
### Tell us more.

Could we please also add more [ASDF plugins](https://github.com/asdf-vm/asdf-plugins)?

For example:
- yq
- jq
- github-cli
- sbt

As previously done in PRs like:
- https://github.com/renovatebot/renovate/pull/18612
- https://github.com/renovatebot/renovate/pull/24146
- https://github.com/renovatebot/renovate/pull/21942
- https://github.com/renovatebot/renovate/pull/22021
- https://github.com/renovatebot/renovate/pull/21999
- https://github.com/renovatebot/renovate/pull/21765
- https://github.com/renovatebot/renovate/pull/21306

**Comment by rarkins** [comment]
PRs welcome

**Reply by MPV** [reply]
Would it be an option to change how the ASDF manager works, so all manual PRs (like above) aren't required? 

For example by making use of something like:

### 1. ASDF community plugins directory

See:
- https://github.com/asdf-vm/asdf-plugins

As they describe it:
>Plugins listed here should be stable and actively maintained. If you have issues with a specific plugin please raise them on the plugin repository first. If a deprecated plugin is listed here, please let us know and create a PR to add the most used alternative.

### 2. `asdf-plugin-manager`

See:
- https://github.com/asdf-community/asdf-plugin-manager
- https://github.com/renovatebot/renovate/discussions/24667

As they describe it:
>#### [Why?](https://github.com/asdf-community/asdf-plugin-manager#why)
>[Asdf is a great universal version manager](https://tech.aabouzaid.com/2022/01/asdf-vm-a-universal-version-manager-tools.html). However, it lacks a secure and declarative method to manage its plugins. For example, you cannot pin a specific asdf plugin version, which means you will be easily hacked if one of the asdf plugins you use is compromised!
>
>https://github.com/asdf-vm/asdf/issues/1577, but no solution has been proposed in asdf upstream yet! (Last check: Aug 2023)
>
>Hence, asdf-plugin-manager fills the gap to manage asdf plugins securely and declaratively via .plugin-versions file.

**Reply by rarkins** [reply]
We could potentially have an asdf-plugins datasource which matches the plug-in name to that public repo and then proxies the lookup to the applicable datasource (Eg github-tags)

**Reply by rarkins** [reply]
Alternatively we could use it as another static data source

**Reply by MPV** [reply]
Or we could read from the `.plugins-versions` file (if it exists), and use those repos and revisions for retrieving dependencies, as per:
- https://github.com/renovatebot/renovate/discussions/24667

Benefit of this approach include:
- reproducibility. 
- safety (only trusted/allowlisted revisions of repos would be used).

Though as asdf-plugin-manager isn't built-in to asdf yet, a fallback method will be needed too (possibly the plugins repo).
